### ISSUE TITLE
`UP018` doesn't detect integers literals used with unary operators

### ISSUE THREAD

**Issue by dosisod** [astral-sh/ruff#10029]
UP018 doesn't warn about integer literals when prefixed with unary operators:

```python
_ = int(1)
_ = int(-1)
_ = int(+1)
_ = int(~1)
```
Running:
```
$ ruff --version
ruff 0.2.2

$ ruff file.py --output-format=concise
file.py:1:5: UP018 [*] Unnecessary `int` call (rewrite as a literal)
```

Only the first line is detected, but all 4 lines should emit errors.

**Comment by charliermarsh** [comment]
Agreed.

**Comment by sanxiyn** [comment]
I am not sure about `~1`. It is bad, but `~1` is not a literal, so I think it is out of scope for UP018.

**Comment by trag1c** [comment]
> I am not sure about `~1`. It is bad, but `~1` is not a literal, so I think it is out of scope for UP018.

Why is `-1` a literal but `~1` isn't? Both have the same AST structure

**Comment by sanxiyn** [comment]
I mean, `-1` is not a literal either in terms of AST, but you could say it is a literal in spirit, since it evaluates to itself. That doesn't apply to `~1`, which evaluates to `-2`. If `~1` is a literal, `60 * 60` is a literal too.
